### ISSUE TITLE
[6.2.2] Creating Watchers directly in ES / what indexes are used

### ISSUE THREAD

**Issue by myrklewk** [sirensolutions/sentinl#370]
I would like to create watchers from an external application using the ES REST api. I found other closed issues which suggested writing to the appropriate indexes directly where sentinl would pick them up.

Problem is, I don't see any indexes associated with sentinl watchers in ES. I can create watchers through the Kibana UI, they run, alerts are created, I see the watchers_alarms index, but no index containing the watchers themselves.
What am I missing?

**Comment by lmangani** [comment]
depending on  your configuration settings, by default `watcher` index is used to store watchers, and `watcher_alarms` to store execution results - You can query those directly from Kibana and see their contents which should be 1:1 with the programmed watcher you saved - You can manually edit and insert from Kibana or other scripts as long as you respect the common format. Sentinl will load up and schedule/remove any new alerts each :25 and :55 

Let me know if this helps.

**Comment by myrklewk** [comment]
So, are you saying that watchers I create through the UI don't get saved in the watcher index but if *I* create one, of the correct format, in the watcher index Sentinl will pick it up? Because I've created numerous watchers through the UI in Kibana, and they execute and generate alarms, but there is no watcher index.

I haven't changed the configuration settings for Sentinl, so everything should be default. What is the default document type? Is it watch (that's what it seems to be in the rookie tutorial in the docs)?

I created a watcher like this (it's a duplicate of one I created in the UI, I just copied the contents of the raw tab as the body of the PUT):

```
PUT watcher/watch/beta
{
  "actions": {
    "email_admin": {
    "throttle_period": "0h0m0s",
      "email": {
        "to": "alarm@localhost",
        "from": "sentinl@localhost",
        "subject": "Alarm",
        "priority": "high",
        "body": "Testtesttest {{payload.hits.total}} Events"
      }
    }
  },
  "input": {
    "search": {
      "request": {
        "index": [ "csm.dt.float.daily_volume_diffs-000001" ],
        "body": {
          "size": 0,
          "query": {
            "constant_score": {
              "filter": {
                "bool": {
                  "must": [
                    {
                      "range": {
                        "@timestamp": {
                          "gte": "now-300s"
                        }
                      }
                    },
                    {
                      "range": {
                        "value": {
                          "gt": 50
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  "condition": {
    "script": {
      "script": "payload.hits.total > 0"
    }
  },
  "transform": {},
  "trigger": {
    "schedule": {
      "later": "every 1 minutes"
    }
  },
  "disable": false,
  "report": false,
  "title": "beta"
}
```
I see it in the index, but it never gets executed, doesn't show up in the Sentinl "Watchers" list, alarms based on it never occur. The exact same watcher created through the UI (with a different name and action message so I can distinguish between them) shows up in the list, executes, and generates alarms.

**Comment by lmangani** [comment]
Check the kibana logs they should tell you why its not being loaded perhaps. I would suggest copying the document from the Dev enviroment rather than copy-paste the raw tab to be accurate in terms of destination format.

**Comment by myrklewk** [comment]
There's nothing obvious in the log, I've attached the start of the log when kibana starts up. Note that the three watchers appearing in the log (two of which are disabled) were all created through the Kibana/Sentinl UI, there is a fourth in the watcher index that doesn't show up in the logs at all (that's the one I created with the PUT).
[klog.txt](https://github.com/sirensolutions/sentinl/files/1839170/klog.txt)

Is Developer mode required to do this sort of thing? I've been using user mode because the server in question is heavily locked down and firewalled and so setting up the dev environment is not straightforward.
